client/main.go

```diff
@@ -41,55 +37,16 @@ func NewFullClient(conn grpc.ClientConnInterface) *FullClient {
 	}
 }
 
-type ConnOptions struct {
-	CertURL       string
-	GRPCAuthority string
-	TLS           bool
-}
-
-func (o *ConnOptions) SetFlags(flagSet *pflag.FlagSet) {
-	flagSet.StringVar(&o.CertURL, "tinkerbell-cert-url", "http://127.0.0.1:42114/cert", "The URL where the certificate is located")
-	flagSet.StringVar(&o.GRPCAuthority, "tinkerbell-grpc-authority", "127.0.0.1:42113", "Connection info for tink-server")
-	flagSet.BoolVar(&o.TLS, "tinkerbell-tls", true, "Connect to server via TLS or not")
-}
-
-// This function is bad and ideally should be removed, but for now it moves all the bad into one place.
-// This is the legacy of packethost/cacher running behind an ingress that couldn't terminate TLS on behalf
-// of GRPC. All of this functionality should be ripped out in favor of either using trusted certificates
-// or moving the establishment of trust in the certificate out to the environment (or running in no-tls mode
-// e.g. for development.)
-func grpcCredentialFromCertEndpoint(url string) (credentials.TransportCredentials, error) {
-	resp, err := http.Get(url)
-	if err != nil {
-		return nil, errors.Wrap(err, "fetch cert")
-	}
-	defer resp.Body.Close()
-
-	certs, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, errors.Wrap(err, "read cert")
+func NewClientConn(authority string, tls bool) (*grpc.ClientConn, error) {
```

Comment:
Can we change this to "target" instead of "authority"?  [In gRPC, "authority"](https://pkg.go.dev/google.golang.org/grpc#WithAuthority) seems to be equivalent to an HTTP "Host" header. In most cases its the same as the target, but can be different.

```suggestion
func NewClientConn(target string, tls bool) (*grpc.ClientConn, error) {
```

Reply:
I don't have a strong opinion here, just wanted to bring it up